jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/JdbcRecordIterator.java

```diff
@@ -58,10 +62,29 @@ public JdbcRecordIterator(Connection conn, PreparedStatement ps, ResultSet rs, C
     if (conf.get(Constants.JDBC_TABLE) != null && conf.get(Constants.JDBC_QUERY) != null) {
       fieldNamesProperty = Preconditions.checkNotNull(conf.get(Constants.JDBC_QUERY_FIELD_NAMES));
       fieldTypesProperty = Preconditions.checkNotNull(conf.get(Constants.JDBC_QUERY_FIELD_TYPES));
-    } else {
-      fieldNamesProperty = Preconditions.checkNotNull(conf.get(serdeConstants.LIST_COLUMNS));
+    }
+    else {
```

Comment:
Nitpicking here again sorry, lets move the else a line up `} else {`

Also regarding else without the `{ ... }` being valid, you are correct it is valid java. Stylistically however using the brackets tends to be more readable and seems to be the dominant style in this project. :+1:

Reply:
fixed.